src/app/api/core/utils/withRetry.ts

```diff
@@ -0,0 +1,49 @@
+import { StatusableError } from '@/types/CopilotApiError'
+import pRetry, { FailedAttemptError } from 'p-retry'
+import * as Sentry from '@sentry/nextjs'
+
+export const withRetry = async <T>(fn: (...args: any[]) => Promise<T>, args: any[]): Promise<T> => {
+  let isEventProcessorRegisterd = false
+
+  return await pRetry(
+    async () => {
+      try {
+        return await fn(...args)
+      } catch (error) {
+        // Hopefully now sentry doesn't report retry errors as well. We have enough triage issues as it is
+        Sentry.withScope((scope) => {
+          if (isEventProcessorRegisterd) return
+
+          isEventProcessorRegisterd = true
```

Comment:
Is this a typo? `isEventProcessorRegisterd` should be `isEventProcessorRegistered`.